packages/core/content-manager/server/services/entity-manager.js

```diff
@@ -206,27 +206,27 @@ module.exports = ({ strapi }) => ({
     return strapi.entityService.deleteMany(uid, params);
   },
 
-  publish: wrapWithEmitEvent(ENTRY_PUBLISH, async (entity, uid) => {
+  publish: wrapWithEmitEvent(ENTRY_PUBLISH, async (entity, body = {}, uid) => {
```

Comment:
Why add this body logic here instead of modifying the updated_by field directly from the caller (aka modifying the entity)?

Reply:
because it woul dbe an update not a publish and we need to emit event. We use this on both collection and single type that's just to avoid duplication or doing too many db queries